infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -94,23 +98,20 @@
      * @param noVerify      Disable checksum if true
      */
     public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
-        logDir = serverContext.getServerConfig().get("--log-path") + File.separator + "log";
-        File dir = new File(logDir);
-        if (!dir.exists()) {
-            dir.mkdirs();
-        }
-
+        logDir = Paths.get(serverContext.getServerConfig().get("--log-path").toString(), "log");
         writeChannels = new ConcurrentHashMap<>();
         channelsToSync = new HashSet<>();
         this.verify = !noVerify;
-        this.serverContext = serverContext;
+        this.dataStore = StreamLogDataStore.builder().dataStore(serverContext.getDataStore()).build();
+
+        initStreamLogDirectory();
         verifyLogs();
         // Starting address initialization should happen before
         // initializing the tail segment (i.e. initializeMaxGlobalAddress)
-        startingAddress = serverContext.getStartingAddress();
+        long startingAddress = dataStore.getStartingAddress();
```

Comment:
Why did you drop the volatile from startingAddress?

Reply:
I don't use `startingAddress` and `tailSegment` in StreamLogFiles, I moved them to StreamLogDataStore. They are volatile in StreamLogDataStore